jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamInitRunner.kt

```diff
@@ -9,46 +9,59 @@ import com.intellij.openapi.vfs.VfsUtil
 import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.util.text.SemVer
 import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.jetbrains.services.lambda.TemplateParameters
 import software.aws.toolkits.jetbrains.services.lambda.sam.SamCommon
 import software.aws.toolkits.resources.message
 
 object SamInitRunner {
+    private val LOG = getLogger<SamInitRunner>()
+
     fun execute(
         name: String,
         outputDir: VirtualFile,
         runtime: Runtime,
-        location: String? = null,
-        dependencyManager: String? = null
+        templateParameters: TemplateParameters
     ) {
+        val isSamGte30 = SemVer.parseFromText(SamCommon.getVersionString())
+            ?.isGreaterOrEqualThan(0, 30, 0) ?: false
+
         // set output to a temp dir
         val tempDir = createTempDir()
         val commandLine = SamCommon.getSamCommandLine()
             .withParameters("init")
             .withParameters("--no-input")
-            .withParameters("--name")
-            .withParameters(name)
-            .withParameters("--runtime")
-            .withParameters(runtime.toString())
             .withParameters("--output-dir")
             .withParameters(tempDir.path)
             .apply {
-                location?.let {
-                    this.withParameters("--location")
-                        .withParameters(it)
-                }
+                when (templateParameters) {
+                    is TemplateParameters.AppBasedTemplate -> {
+                        this.withParameters("--name")
+                            .withParameters(name)
+                            .withParameters("--runtime")
+                            .withParameters(runtime.toString())
+                            .withParameters("--dependency-manager")
+                            .withParameters(templateParameters.dependencyManager)
 
-                dependencyManager?.let {
-                    this.withParameters("--dependency-manager")
-                        .withParameters(it)
+                        if (isSamGte30) {
```

Comment:
Like the previous SAM change like this, we should add a `FIX_WHEN_SAM_MIN_IS_0_30` or similar